### ISSUE TITLE
LogEntryEvent__e.RequestId__c field is not being set

### ISSUE THREAD

**Issue by chazwatkins** [jongpie/NebulaLogger#424]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

v4.9.7

### New Bug Summary

We upgraded from 4.7.0 to 4.9.7 yesterday and are getting errors from an automation user stating the LogEntryEvent__e.RequestId__c must have a value.  The user has all Logger perm sets, which have access to RequestId__c.

The v4.8.4 release notes, where RequestId__c was introduced, don't note any changes required post-installation to utilize RequestId__c.

Any idea what could be causing this error?

__Error message__
```
Logger failed to save 1 LogEntryEvent__e records for ...

Error Messages:

You must enter a value: RequestId__c
```

For a temporary fix, we've set RequestId__c as not a required field.

**Comment by jongpie** [comment]
Hey @chazwatkins thanks for reporting this! The field `RequestId__c` shouldn't have been flagged as required - I seldomly add fields that should be required, and this field being marked as required was unintentional - I update it in the next release so that it's no longer required.

However, I'm curious if you were only getting this error some of the time - the field is [currently being set](https://github.com/jongpie/NebulaLogger/blob/f7359592b783543d39072eac6dffe91fe70df9aa/nebula-logger/core/main/logger-engine/classes/Logger.cls#L2939) to be the value of `System.Request.getCurrent().getRequestId()`, so I'm wondering if perhaps it's only null in some situations? Does it sometimes work for you?

**Comment by chazwatkins** [comment]
Hey @jongpie - Thanks for getting back to me so quickly.  The error only appears when we fire a platform event and the subscriber transaction, processed using the autoproc user, uses Logger.

**Comment by jongpie** [comment]
@chazwatkins thanks for the info! That's interesting, I guess it makes sense that the autoproc user doesn't have a request ID, but that wasn't something I had tested before.

I should have the next release `v4.9.8` (PR #423) released sometime this week, which will include the change to make the field `LogEntryEvent__e.RequestId__c` optional.

**Comment by jongpie** [comment]
Hey @chazwatkins - this should now be fixed in the newest release [v4.9.8](https://github.com/jongpie/NebulaLogger/releases/tag/v4.9.8). Thanks again for reporting this, and let me know if you run into any other issues!
